### DISCUSSION TITLE
CMS Tall CSV to large for website. Is it okay to post a ZIPPED version of the file?

### DISCUSSION THREAD

**Discussion by LoganPascoal** [CMSgov/hospital-price-transparency#118] [Q&A] [answered]
My tall files are upwards of 1.4 million KBs. Our system has a limit and even then I am worried about the load time impact on our servers and site.  Is it against CMS best practice to compress the file into a Zip folder to post online?

**Comment by RidenourK** [comment, marked as answer]
@LoganPascoal. Thank you for your question. 

The Hospital Price Transparency Rule does not restrict hospitals from zipping, or compressing for storage, the machine-readable file. We recommend if a hospital chooses to zip or compress a file that the hospital use the CMS-specified naming convention for the file (Refer to § 180.50(d)(5)).

**Reply by nmaylevc** [reply]
Two questions about the response:

1. when you say "use the CMS-specified naming convention", since the rule says the file extension must be "[json|csv]" how is this requirement modified by compressing the file?  Should the .zip or .gz be appended after the .json or .csv or should they replace that extension?
2. when you say "recommend" do you mean that this is a suggestion or a requirement/guidance?  The [rule language](https://www.ecfr.gov/current/title-45/part-180/section-180.50#p-180.50(d)(5)) itself is fairly clear (bold added):

> The machine-readable file **must use** the following naming convention specified by CMS, specifically: <ein>_<hospital-name>_standardcharges. [json|csv].

We see some hospitals leaving off the .json or .csv and some not.

**Reply by RidenourK** [reply]
@nmaylevc. Thank you for your questions. 

CMS recommends, but does not require, hospitals to use the CMS-specified naming convention for zipped files. For hospitals that choose to use the CMS specified naming convention for a zipped file, the .zip or .gz should be appended after the .json or .csv.

**Reply by nmaylevc** [reply]
Why would the naming convention not be required for a zipped version of the MRF? The rule is quite clear that the naming convention is required for the MRF.

**Reply by RidenourK** [reply]
@nmaylevc. Thank you for your question. 

Are you (or other innovators who wish to comment on this post) using the CMS-specified naming convention to locate MRFs?

**Reply by nmaylevc** [reply]
We are using the .txt files, the footer and the naming convention to locate MRFs.
On many hospital websites the pages that contain MRFs also contain links to other similar documents and we need to determine which is which.  The .txt file is a big help, but it isn't always there and when it is, it doesn't always contain the link to the latest MRFs.  So, we use all the hints that are available.  A big weakness of the naming convention is that it is the "file" naming convention and not a naming convention for the URL.

**Reply by JamJomJim** [reply]
Following up here. I'm also relying on the .txt and file naming conventions, and am also confused that -- while the rule very specifically states that the naming convention is required -- this is only _suggested_ here. Thanks for your previous responses @RidenourK, but are you able to answer @nmaylevc questions? It'd help me along with others that look at this post for clarity.
